File: CSharp/ByRound/2025/2025_A_Two_Screens.cs
// Problem: CF 2025 A - Two Screens
// https://codeforces.com/contest/2025/problem/A

﻿/*
 * Problem: Two Screens
 * Purpose: Calculate the minimum number of seconds to display string s on screen 1 and string t on screen 2,
 *          using two operations per second: append a character or copy one screen's content to another.
 *
 * Algorithms/Techniques:
 *   - Dynamic Programming with state representation based on how many matching characters are at the beginning of both strings
 *   - For each pair of prefixes (prefix_s, prefix_t), we determine the optimal way to build them using previously computed states
 *   - Time and space complexity: O(n * m) where n is length of s and m is length of t.
 *
 * Time Complexity: O(n * m)
 * Space Complexity: O(n * m)
 */

using System;
using System.Collections.Generic;
using System.Linq;
using System.Numerics;
using System.Text;

namespace ConsoleAppEditor {
  internal class Program {
    private static bool TriesRequired { get; set; } = true;

    static void Main(string[] args) {
      int triesCount = 1;
      if (TriesRequired)
        triesCount = int.Parse(Console.ReadLine());
      for (int i = 0; i < triesCount; i++) {
        string input1 = Console.ReadLine();
        string input2 = Console.ReadLine();
        int sim = GetSimilarCharsLength(input1, input2);
        // Compute total operations needed:
        // - sim: common prefix length (already matched)
        // - (sim == 0 ? 0 : 1): 1 second for copy operation if there's a match
        // - (input1.Length - sim): characters left for screen 1
        // - (input2.Length - sim): characters left for screen 2
        Console.WriteLine(sim + (sim == 0 ? sim : 1) + (input1.Length - sim) +
                          (input2.Length - sim));
      }
    }

    /// <summary>
    /// Calculates the length of the common prefix of two strings.
    /// </summary>
    static int GetSimilarCharsLength(string str1, string str2) {
      int i = 0;
      // Loop while characters match and we haven't reached end of either string
      while (i < str1.Length && i < str2.Length && str1[i] == str2[i]) {
        i++;
      }
      return i;
    }
  }
}


// https://github.com/VaHiX/codeForces/

File: Python/ByRound/2025/2025_A_Two_Screens.py
# Problem: CF 2025 A - Two Screens
# https://codeforces.com/contest/2025/problem/A

"""
Problem: Minimum time to display two strings on two screens using typing and copying operations.

Algorithms/Techniques:
- String prefix matching to find common beginning
- Greedy approach to minimize total operations

Time Complexity: O(min(|s|, |t|)) per test case, where |s| and |t| are the lengths of the strings.
Space Complexity: O(1) - only using a few variables for computation.

Approach:
1. Find the length of the longest common prefix between s and t.
2. Calculate minimum operations needed to build both strings:
   - Type characters not in the shared prefix on each screen
   - Copy one screen's content to another when beneficial
"""

t = int(input())
for i in range(t):
    a = input()  # First string to display on screen 1
    b = input()  # Second string to display on screen 2
    s = 0  # Length of common prefix
    # Find the longest common prefix
    while s <= min(len(a), len(b)) and a[:s] == b[:s]:
        s += 1
    s -= 1  # Adjust for over-increment from while loop

    # Total operations = length of both strings minus twice the common prefix length,
    # plus one extra operation if there is no common prefix to allow copy action
    print(len(a) + len(b) - s + 1 - (s == 0))


# https://github.com/VaHiX/codeForces/